From 9b478d607922c9683663f8e317d93b840e528884 Mon Sep 17 00:00:00 2001
From: Dmitry Eremin <dmitry.eremin@intel.com>
Date: Thu, 5 May 2016 22:08:05 +0300
Subject: [PATCH 1/8] LU-8056 libcfs: Support for linux 4.2 kernels

In kernel version 4.2 commit df6b35f409af0a8ff1ef62f552b8402f3fef8665
header file i387.h was renamed to fpu/api.h

Change-Id: Id4f5f6e73f3c2aeac67b5c87b9c1d310a0a50a50
Signed-off-by: Dmitry Eremin <dmitry.eremin@intel.com>
---
 libcfs/autoconf/lustre-libcfs.m4                 | 11 +++++++++++
 libcfs/libcfs/linux/linux-crypto-crc32c-pclmul.c |  4 ++++
 libcfs/libcfs/linux/linux-crypto-crc32pclmul.c   |  4 ++++
 3 files changed, 19 insertions(+)

diff --git a/libcfs/autoconf/lustre-libcfs.m4 b/libcfs/autoconf/lustre-libcfs.m4
index 636d0ed..cc0e184 100644
--- a/libcfs/autoconf/lustre-libcfs.m4
+++ b/libcfs/autoconf/lustre-libcfs.m4
@@ -312,6 +312,16 @@ topology_sibling_cpumask, [
 ]) # LIBCFS_HAVE_TOPOLOGY_SIBLING_CPUMASK
 
 #
+# Kernel version 4.2 commit df6b35f409af0a8ff1ef62f552b8402f3fef8665
+# header file i387.h was renamed to fpu/api.h
+#
+AC_DEFUN([LIBCFS_FPU_API], [
+LB_CHECK_LINUX_HEADER([asm/fpu/api.h], [
+	AC_DEFINE(HAVE_FPU_API_HEADER, 1,
+		[fpu/api.h is present])])
+]) # LIBCFS_FPU_API
+
+#
 # LIBCFS_PROG_LINUX
 #
 # LibCFS linux kernel checks
@@ -350,6 +360,7 @@ LIBCFS_SHRINKER_COUNT
 LIBCFS_HLIST_ADD_AFTER
 # 4.2
 LIBCFS_HAVE_TOPOLOGY_SIBLING_CPUMASK
+LIBCFS_FPU_API
 ]) # LIBCFS_PROG_LINUX
 
 #
diff --git a/libcfs/libcfs/linux/linux-crypto-crc32c-pclmul.c b/libcfs/libcfs/linux/linux-crypto-crc32c-pclmul.c
index 9858db4..fc55ad7 100644
--- a/libcfs/libcfs/linux/linux-crypto-crc32c-pclmul.c
+++ b/libcfs/libcfs/linux/linux-crypto-crc32c-pclmul.c
@@ -30,7 +30,11 @@
 #include <crypto/internal/hash.h>
 #include <linux/crc32.h>
 #include <asm/cpufeature.h>
+#ifdef HAVE_FPU_API_HEADER
+#include <asm/fpu/api.h>
+#else
 #include <asm/i387.h>
+#endif
 #include <libcfs/libcfs.h>
 
 #define CHKSUM_BLOCK_SIZE	1
diff --git a/libcfs/libcfs/linux/linux-crypto-crc32pclmul.c b/libcfs/libcfs/linux/linux-crypto-crc32pclmul.c
index 1a609bf..0b3abaf 100644
--- a/libcfs/libcfs/linux/linux-crypto-crc32pclmul.c
+++ b/libcfs/libcfs/linux/linux-crypto-crc32pclmul.c
@@ -32,7 +32,11 @@
 #include <crypto/internal/hash.h>
 #include <linux/crc32.h>
 #include <asm/cpufeature.h>
+#ifdef HAVE_FPU_API_HEADER
+#include <asm/fpu/api.h>
+#else
 #include <asm/i387.h>
+#endif
 #include <libcfs/libcfs.h>
 
 #define CHKSUM_BLOCK_SIZE	1
-- 
2.8.2

